sklearn/neural_network/rbm.py

```diff
@@ -1,367 +1,12 @@
-"""Restricted Boltzmann Machine
-"""
+import warnings
 
-# Authors: Yann N. Dauphin <dauphiya@iro.umontreal.ca>
-#          Vlad Niculae
-#          Gabriel Synnaeve
-#          Lars Buitinck
-# License: BSD 3 clause
+from ._rbm import *  # noqa
 
-import time
 
-import numpy as np
-import scipy.sparse as sp
-from scipy.special import expit  # logistic function
+msg = ("The `sklearn.neural_nework.rbm` module is  deprecated in version "
+       "0.22 and will be removed in version 0.24. "
+       "The corresponding classes / functions "
+       "should instead be imported from sklearn.neural_nework")
```

Comment:
related to the pickle backward incompatibility raised by @jnothman in https://github.com/scikit-learn/scikit-learn/issues/12927#issuecomment-529240290, one option is to tell the users that they are moved to `_*` instead, are private, and anything they use from that path and not the usual public path, can be changed without warning. This way we at least tell them they have the option, if they take the responsibility themselves. WDYT?

Reply:
I think the pickle is only an issue if users have been importing from `sklearn.sub_package.submodule.py` instead of importing from `sklearn.sub_package`. In this case I don't think we want to tell them that they can import from `sklearn.sub_package._sub_module.py`.

Would you be OK with simply adding e.g. "Anything that cannot be imported from sklearn.neural_network is now part of the private API."

Reply:
yeah, I'd be happy with that.